File: rps.py
#Python program to play rock, paper and scissors

from cs50 import get_string
import random

def game():
    playerWins, compWins = 0, 0

    while playerWins < 5  and compWins < 5:
        yourTurn = get_string("\nYour Turn: ")

        compTurn = random.choice(["rock", "paper", "scissor"])

        print(f"\nComp's Turn: {compTurn}")

        if (yourTurn == "rock"  and  compTurn == "paper")  or  (yourTurn == "paper"  and compTurn == "scissor")  or  (yourTurn == "scissor"  and  compTurn == "rock"):
            compWins += 1
        elif (yourTurn == "rock"  and  compTurn == "scissor")  or  (yourTurn == "paper"  and compTurn == "rock")  or  (yourTurn == "scissor"  and  compTurn == "paper"):
            playerWins += 1

        print(f"\n\nYour Wins: {playerWins}\t\tComp's Wins: {compWins}")

    if playerWins == 5:
        print("\nYou Win!")
    else:
        print("\nComputer Wins!")

def main():
    print("Let's play Rock, Paper and Scissors")
    print("\nYou have to win 5 times first to win. And you have to type in what you choose")

    print("\n\n\t\tDEMO")
    print("\nYour turn: rock\t\tComputer's Turn: paper")
    print("\nComputer wins!\n")

    play = "y"

    while play == "y":
        game()
        play = get_string("\nDo you wanna play again?(y/n)")
        print("Okay")

if __name__ == "__main__":
    main()

File: rps.c
//Program to play rock, paper and scissors

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>
#include <stdbool.h>
#include <cs50.h>

void game();

int main(void)
{

    printf("Let's play Rock, Paper and Scissors\n");
    printf("\nYou have to win 5 times first to win. And you have to type in what you choose\n");

    printf("\n\n\t\tDEMO\n");
    printf("\nYour turn: rock\t\tComputer's Turn: paper\n");
    printf("\nComputer wins\n");

    char play = 'y';

    do
    {
        game();
        printf("\nDo you wanna play again?");
        scanf("%s", &play);
        printf("okay\n");

    }while(play != '0');
}

void game()
{
    int playerWins = 0, compWins= 0;

    while(playerWins < 5  &&  compWins < 5)
    {
        char yourTurn[9], compTurn[9];
        int comp;

        printf("\nYour turn: ");
        scanf("%s", yourTurn);

        printf("\nComputer's Turn: ");

        srand(time(0));
        comp = rand() % 3;

        switch(comp)
        {
            case 0: strcpy(compTurn, "rock");
                    printf("%s\n", compTurn);
                    break;
            case 1: strcpy(compTurn, "paper");
                    printf("%s\n", compTurn);
                    break;
            case 2: strcpy(compTurn, "scissor");
                    printf("%s\n", compTurn);
                    break;
        }

        if((!strcmp(yourTurn, "rock")  &&  !strcmp(compTurn, "scissor"))  ||  (!strcmp(yourTurn, "paper")  &&  !strcmp(compTurn, "rock"))  ||  (!strcmp(yourTurn, "scissor")  &&  !strcmp(compTurn, "paper")))
        {
            playerWins++;
        }
        else if((!strcmp(yourTurn, "rock")  &&  !strcmp(compTurn, "paper"))  ||  (!strcmp(yourTurn, "paper")  &&  !strcmp(compTurn, "scissor"))  ||  (!strcmp(yourTurn, "scissor")  &&  !strcmp(compTurn, "rock")))
        {
            compWins++;
        }

        printf("\n\nYour Wins: %i\t\tComp's Wins: %i\n", playerWins, compWins);
    }

    if(playerWins == 5)
        printf("\nYou Win!\n");
    else
        printf("\nComputer Wins!\n");
}